### PR TITLE
Water bombs stun instead of kill when they explode in your inventory

### PR DESCRIPTION
**Pull request by eps0003**
## Status

**READY**

## Description

They also stun the area around you as water bombs do

## Steps to Test or Reproduce

Let a water bomb explode in your inventory

### CODE REVIEW

File: Entities/Items/Explosives/Explosion.as [hunk 1]
```diff
@@ -69,10 +69,17 @@ void Explode(CBlob@ this, f32 radius, f32 damage)
 		{
 			//copy position, explode from centre of carrier
 			pos = doomed.getPosition();
-			//kill players if we're in their inventory (even water bombs for now)
+			//kill players if we're in their inventory
```
Comment (AsuMagic) [inline Entities/Items/Explosives/Explosion.as]: That comment is now misleading.

File: Entities/Items/Explosives/Explosion.as [hunk 2]
```diff
@@ -69,10 +69,17 @@ void Explode(CBlob@ this, f32 radius, f32 damage)
...
 			if ((doomed.hasTag("player") || doomed.getName() == "crate") && !doomed.hasTag("invincible"))
 			{
-				this.server_Hit(doomed, pos, Vec2f(), 100.0f, Hitters::explosion, true);
+				if (this.getName() == "bomb")
+				{
+					this.server_Hit(doomed, pos, Vec2f(), 100.0f, Hitters::explosion, true);
+				}
+				else if (this.getName() == "waterbomb")
+				{
+					this.server_Hit(doomed, pos, Vec2f(), 0.0f, Hitters::water_stun, true);
```
Comment (AsuMagic) [inline Entities/Items/Explosives/Explosion.as]: Maybe `water_stun_force` instead? Not sure if that would make much of a difference.

Comment (eps0003) [reply Entities/Items/Explosives/Explosion.as]: It doesn't seem to make any impact

Comment (AsuMagic) [reply Entities/Items/Explosives/Explosion.as]: Even when shielding and with sponges in inventory?

Comment (eps0003) [reply Entities/Items/Explosives/Explosion.as]: I didn’t think to test those

**Comment by furai** [comment]
@eps0003 what about that case mentioned by @AsuMagic? Did you ever get to test it properly?

**Pull request merged**